server/github/builder.go

```diff
@@ -34,3 +37,40 @@ func ImageFromTemplate(t *template.Template) ImageBuilder {
 		return image.Decode(buf.String())
 	})
 }
+
+// conveyorClient mocks the interface to the Conveyor API.
+type conveyorClient interface {
+	Build(io.Writer, conveyor.BuildCreateOpts) (*conveyor.Artifact, error)
+}
+
+// ConveyorImageBuilder provides an ImageBuilder implementation that
+// integrations with the Conveyor (https://github.com/remind101/conveyor) Docker
+// build system. If enabled, Empire will check if an Artifact in Conveyor exists
+// for the git commit, and will trigger Conveyor to build it if it doesn't
+// exist.
+type ConveyorImageBuilder struct {
+	client conveyorClient
+}
+
+// NewConveyorImageBuilder returns a new ConveyorImageBuilder implementation
+// that uses the given client.
+func NewConveyorImageBuilder(c *conveyor.Service) *ConveyorImageBuilder {
+	return &ConveyorImageBuilder{
+		client: c,
+	}
+}
+
+func (c *ConveyorImageBuilder) BuildImage(ctx context.Context, event events.Deployment) (image.Image, error) {
+	w := ioutil.Discard // TODO: Allow ImageBuilder implementations to write to the deployment logs?
+
+	a, err := c.client.Build(w, conveyor.BuildCreateOpts{
+		Repository: event.Repository.FullName,
+		Branch:     "", // TODO,
```

Comment:
This should _probably_ be `event.Deployment.Ref`. Ultimately, Conveyor will need to support "branchless" builds, which should be pretty easy since the branch is only used to optimize the layer cache and isn't explicitly necessary to perform a build.

This is where things get a little weird, since we don't actually know if the ref that the deployment was triggered for was a branch, tag or sha.